features/unit/v2algodclient_paths.feature

```diff
@@ -81,6 +81,15 @@ Feature: Algod REST API v2 Paths
       | /v2/applications/1234/box?name=b64%3A8J%2BSqQ%3D%3D | 1234           | b64:8J+SqQ==         |
       | /v2/applications/1234/box?name=b64%3AYS96           | 1234           | b64:YS96             |
 
+  @unit.applications.boxes
+  Scenario Outline: GetApplicationBoxes
+    When we make a GetApplicationBoxes call for applicationID <application-id> with max <max>
```

Comment:
@algochoi No problem - Arguably it's more correct to exercise an example with all parameters.

@algochoi I've made corresponding Go + Java changes though the Java change is questionable.  Can you have a read through and let me know what you think?
  
* The Java generator does _not_ omit `max=0`, so the test fails though the SDK-produced URL is valid.
* I made a change I'm uncomfortable with to pass the test:  https://github.com/algorand/java-algorand-sdk/pull/347/commits/7d7c08d2fa5d40441319972f610ef2750a331e1c.
* Open to discussing verbally:  Although the tests pass, my preference is to remove the max = 0 test case.
  * I think it'd be preferable to have less coverage than retain the change.
  * I prefer to avoid a _generator_ templating change to support the case.
  * It's possible we can come up with a test case representation that's compatible with Java SDK.  I'm open to a brief discussion though I didn't immediately spot an alternative.

Reply:
I agree with your comments - I'm wondering if there is an existing workaround to this as there are other paths that test `max=0` params and omit it in the paths there (haven't looked to deeply myself):
https://github.com/algorand/java-algorand-sdk/blob/7d7c08d2fa5d40441319972f610ef2750a331e1c/src/test/java/com/algorand/algosdk/unit/AlgodPaths.java#L27 

This is the corresponding testcase: 
https://github.com/algorand/algorand-sdk-testing/blob/094764e214d1461bda4c47feaf237256c09f758e/features/unit/v2algodclient_paths.feature#L16

Reply:
@algochoi Thanks for the suggestion - It's implemented in https://github.com/algorand/java-algorand-sdk/pull/347/commits/8fdcef2f98e1034a0ccb0c8c08afccca1e808015.  Resolving and leaving informational context below.

As we discussed verbally, https://github.com/algorand/java-algorand-sdk/pull/347/commits/8fdcef2f98e1034a0ccb0c8c08afccca1e808015 is an improvement.  Though it feels improper to modify the test setup.  Since it's a prevailing convention, we decided it's preferable to retain the test case with a less murky Java SDK workaround.